extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReaderImpl.java

```diff
@@ -264,24 +266,21 @@ public ColumnPageReader next() {
                         throw new UncheckedDeephavenException(
                                 "Unknown parquet data page header type " + pageHeader.type);
                 }
-                final Supplier<Dictionary> pageDictionarySupplier =
-                        (encoding == PLAIN_DICTIONARY || encoding == RLE_DICTIONARY)
-                                ? dictionarySupplier
-                                : () -> NULL_DICTIONARY;
-                return new ColumnPageReaderImpl(
-                        channelsProvider, decompressor, pageDictionarySupplier,
+                if ((encoding == PLAIN_DICTIONARY || encoding == RLE_DICTIONARY)
+                        && dictionarySupplier.get() == NULL_DICTIONARY) {
```

Comment:
This is probably a bad idea. I'd need to look at the dictionary suppliers, but they do a lot of work, and I don't recall if they cache internally.